cfgmgr/natmgrd.cpp

```diff
@@ -62,21 +62,31 @@ NatMgr    *natmgr = NULL;
 NotificationConsumer   *timeoutNotificationsConsumer = NULL;
 NotificationConsumer   *flushNotificationsConsumer = NULL;
 
+static volatile sig_atomic_t gExit = 0;
+
 std::shared_ptr<swss::NotificationProducer> cleanupNotifier;
 
 void sigterm_handler(int signo)
+{
+    SWSS_LOG_ENTER();
+
+    gExit = 1;
+}
+
+int cleanup()
 {
     int ret = 0;
     std::string res;
     const std::string conntrackFlush            = "conntrack -F";
 
-    SWSS_LOG_NOTICE("Got SIGTERM");
+    SWSS_LOG_ENTER();
 
     /*If there are any conntrack entries, clean them */
     ret = swss::exec(conntrackFlush, res);
     if (ret)
     {
         SWSS_LOG_ERROR("Command '%s' failed with rc %d", conntrackFlush.c_str(), ret);
+        return ret;
```

Comment:
>return ret; [](http://example.com/codeflow?start=8&length=11)

For cleanup, it should try best to cleanup everything. If one task failed, it should continue with others.

Reply:
@qiluo-msft done